### PR TITLE
Enforced more Mocha lint rules

### PR DESCRIPTION
**Pull request by nmizoguchi**
ref https://github.com/TryGhost/Ghost/issues/11038

1. Enforced lint rule **[ghost/mocha/no-identical-title](https://github.com/lo1tuma/eslint-plugin-mocha/blob/main/docs/rules/no-identical-title.md)** - Fixed relevant tests
2. Enforced lint rule **[ghost/mocha/max-top-level-suites](https://github.com/lo1tuma/eslint-plugin-mocha/blob/main/docs/rules/max-top-level-suites.md)** - No required fixes, as tests are compliant already

#### Additional details
Specifically for `ghost/mocha/no-identical-title` most fixes were simple test description updates. Added comments to aid the PR review for the ones that had relevant changes, and might require more attention. They are as follows:
* [e2e-api/admin/invites.test.js](https://github.com/TryGhost/Ghost/pull/19720#discussion_r1496397548): Removed duplicated test (exact same code on both);
* [e2e-api/admin/members.test.js](https://github.com/TryGhost/Ghost/pull/19720#discussion_r1496399107): From the[ PR this was introduced](https://github.com/TryGhost/Ghost/commit/73466c1c4055cfbfbc4fdfb96705adeadd92a60e#diff-4dbc7e96e356428561085147e00e9acb5c71b58d4c1bd3d9fc9ac30e77c45be0L236-L237) seems like author based his test on an existing one but possibly forgot to rename it;
* [unit/api/canary/utils/serializers/input/pages.test.js](https://github.com/TryGhost/Ghost/pull/19720#discussion_r1496400143): The [page filter](https://github.com/TryGhost/Ghost/pull/14829/files) was removed, so changed the description accordingly;
* [unit/api/canary/utils/serializers/input/posts.test.js](https://github.com/TryGhost/Ghost/pull/19720#discussion_r1496400329): The [page filter](https://github.com/TryGhost/Ghost/pull/14829/files) was removed, so changed the description accordingly;
* [unit/frontend/services/rendering/templates.test.js](https://github.com/TryGhost/Ghost/pull/19720#discussion_r1496402430): Removed duplicated test
* [unit/server/models/post.test.js](https://github.com/TryGhost/Ghost/pull/19720#discussion_r1496403529): the change in [this PR](https://github.com/TryGhost/Ghost/pull/14586/files#diff-c351cb589adefbb886570cfadb33b33eb8fdc12bde1024d1188cd18c165fc5e8L1010) made three tests here mostly the same. Deduplicated them and kept only one.

---
Please include a description of your change & check your PR against this list, thanks!

- [x] There's a clear use-case for this code change, explained below
- [x] Commit message has a short title & references relevant issues
- [x] The build will pass (run `yarn test:all` and `yarn lint`)

We appreciate your contribution!

### CODE REVIEW

File: ghost/core/test/e2e-api/admin/invites.test.js [hunk 1]
```diff
@@ -224,32 +224,5 @@ describe('Invites API', function () {
                 .expect('Cache-Control', testUtils.cacheRules.private)
                 .expect(403);
         });
-
-        it('Can add a new invite by API Key with the Contributor Role', async function () {
```
Comment (nmizoguchi) [inline ghost/core/test/e2e-api/admin/invites.test.js:228]: [To help reviewers] Removed duplicated test (exactly same code as [this one](https://github.com/TryGhost/Ghost/blob/4598a342f030b2899c18fb4b699495504953c3f2/ghost/core/test/e2e-api/admin/invites.test.js#L188-L213));

File: ghost/core/test/e2e-api/admin/members.test.js [hunk 2]
```diff
@@ -554,7 +554,7 @@ describe('Members API', function () {
             });
     });
 
-    it('Can filter by signup attribution', async function () {
+    it('Can filter by conversion attribution', async function () {
```
Comment (nmizoguchi) [inline ghost/core/test/e2e-api/admin/members.test.js:557]: [To help reviewers] From the[ PR this was introduced](https://github.com/TryGhost/Ghost/commit/73466c1c4055cfbfbc4fdfb96705adeadd92a60e#diff-4dbc7e96e356428561085147e00e9acb5c71b58d4c1bd3d9fc9ac30e77c45be0L236-L237) seems like author based his test on an existing one but possibly forgot to rename it;

File: ghost/core/test/unit/api/canary/utils/serializers/input/pages.test.js [hunk 3]
```diff
@@ -30,7 +30,7 @@ describe('Unit: endpoints/utils/serializers/input/pages', function () {
             frame.options.filter.should.eql('(status:published+tag:eins)+type:page');
         });
 
-        it('combine filters', function () {
+        it('only tag filters', function () {
```
Comment (nmizoguchi) [inline ghost/core/test/unit/api/canary/utils/serializers/input/pages.test.js:33]: [To help reviewers] The [page filter](https://github.com/TryGhost/Ghost/pull/14829/files) was removed, so changed the description accordingly;

File: ghost/core/test/unit/api/canary/utils/serializers/input/posts.test.js [hunk 4]
```diff
@@ -57,7 +57,7 @@ describe('Unit: endpoints/utils/serializers/input/posts', function () {
             frame.options.filter.should.eql('(status:published+tag:eins)+type:post');
         });
 
-        it('combine filters', function () {
+        it('only tag filters', function () {
```
Comment (nmizoguchi) [inline ghost/core/test/unit/api/canary/utils/serializers/input/posts.test.js:60]: [To help reviewers] The [page filter](https://github.com/TryGhost/Ghost/pull/14829/files) was removed, so changed the description accordingly;

File: ghost/core/test/unit/frontend/services/rendering/templates.test.js [hunk 5]
```diff
@@ -454,28 +454,6 @@ describe('templates', function () {
             stubs.pickTemplate.calledWith('test', 'path/to/local/test.hbs').should.be.true();
         });
 
-        it('calls pickTemplate for custom routes', function () {
```
Comment (nmizoguchi) [inline ghost/core/test/unit/frontend/services/rendering/templates.test.js:457]: [To help reviewers] Removed duplicated test (exactly same code as [this one](https://github.com/TryGhost/Ghost/blob/4598a342f030b2899c18fb4b699495504953c3f2/ghost/core/test/unit/frontend/services/rendering/templates.test.js#L435-L455));

File: ghost/core/test/unit/server/models/post.test.js [hunk 6]
```diff
@@ -887,60 +887,6 @@ describe('Unit: models/post: uses database (@TODO: fix me)', function () {
                     });
                 });
 
-                it('rejects if changing authors', function (done) {
```
Comment (nmizoguchi) [inline ghost/core/test/unit/server/models/post.test.js:890]: [To help reviewers] The change in [this PR](https://github.com/TryGhost/Ghost/pull/14586/files#diff-c351cb589adefbb886570cfadb33b33eb8fdc12bde1024d1188cd18c165fc5e8L1010) made three tests here mostly the same. Deduplicated them and kept only one.

**Review by nmizoguchi — COMMENTED** [review]
Left a few comments to aid PR review

**Comment by nmizoguchi** [comment]
@kevinansfield apologies for the direct ping, but wanted to be sure this PR won't get lost!

**Comment by daniellockyer** [comment]
Thanks @nmizoguchi! 🙂

**Pull request merged**